pkg/didcomm/protocol/didexchange/persistence.go

```diff
@@ -86,14 +99,60 @@ func (c *ConnectionRecorder) GetInvitation(verKey string) (*Invitation, error) {
 	return result, nil
 }
 
-// invitationKey computes key for invitation object
-func invitationKey(verKey string) (string, error) {
-	storeKey, err := computeHash([]byte(verKey))
+// SaveConnectionRecorder saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) SaveConnectionRecorder(invitation *Invitation,
+	msg *Message) error {
+	connRecord := &connectionRecord{ThreadID: msg.ThreadID,
+		ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey:          invitation.RecipientKeys, State: msg.State}
+
+	connBytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	return c.store.Put(msg.ConnectionID, connBytes)
+}
+
+// SaveProtocolID saves the hash of the protocol id which is a did and thread id as an ID and connection id as value.
```

Comment:
What will this protocol ID be used for?

Reply:
ProtocolId is a composite key (mydid + thid) and is used to store the connection ID in the store.  Which later on will be used to fetch connection record properties like state.

Reply:
Curious: why is it a hash of myDID + thid? Why can't it just be a UUID?